src/redux/reducers/calls.js

```diff
@@ -0,0 +1,19 @@
+import { LOG_CALL } from "../actionTypes";
+
+const initialState = {
+  byId: {},
+};
+
+export default function(state = initialState, action) {
+  switch (action.type) {
+    case LOG_CALL: {
+      const { districtId, timestamp } = action.payload;
+      return {
+        ...state,
+        byId: {...state.byId, [districtId]: timestamp}
```

Comment:
Should we be concerned with someone triggering this `LOG_CALL` a bunch of times (by refreshing or something)? This is currently saving the _latest_ call. Should we save each call separately? Should we only track one call per district per day?

Reply:
If I understand your concern correctly, you're pointing out that this reducer relies on the action getting dispatched appropriately. It naively upserts data without any guards in place modification of the state and it doesn't provide an auditable full history.
That's a great point. While right now, things are getting dispatched appropriately, it won't always be the case. I'll do a check to ensure that there's a max event per district per 24 hour period.